### ISSUE TITLE
Account requests need corresponding authgroups

### ISSUE THREAD

**Issue by torrin47** [USEPA/GeoPlatform-Request-Tool#38]
It looks like only coordinators and delegates have assigned primary authgroups, but we need the external accounts to be assigned to one and only one primary authgroup that isn't inherited from their sponsor (because sponsors/admins often oversee multiple authgroups).  It would be fine for primary authgroup to be set at the project/response level, as it's a reasonable assumption that each project/response would fall under a single authgroup. But one way or the other we need a primary authgroup connected to each external account. The big exception to this is the newish Limited External Collaborator Role:

- name: "Limited External Collaborator"
- id: "MuPlsRnLCR8yeE5H"
- description: "Role for limiting access to organization resources - view only, no editing, no access to org items, group items only."

Users assigned this role should not be assigned to an organizational authgroup, instead we place them in this authgroup:
https://epa.maps.arcgis.com/home/group.html?id=75a06d7649b2426fb2a892f26fc5d1d1
so they don't inherit access to anything that isn't explicitly share with them. It's fine for them to be assigned to other groups as necessary. Speaking of which, we see roles here:
https://request.ercloud.org/api/v1/account/approvals/roles/
but it's not clear where in the interface a user would be assigned to a role.

**Comment by tbock** [comment]
I think we had the role selection in the approval list at one point but was removed for some reason.  @jlherring can probably provide clarity. 

I follow the rest.  Seems like we would need to add another input defining exactly which response/project they are, at least initially, being assigned to.  Currently only sponsors/delegates get that treatment just to control which groups they can assign people to.  I suspect this has been discussed at some level so need @jlherring to comment first and see where this stands with the ER group.  Although... not sure I see an alternative. 

Bottom line, IMO, the sponsor/delegate won't know which auth group to assign people so this needs to be automatic based on other selections... which sounds like is possible.

**Comment by jlherring** [comment]
@tbock Yes - we had the ability to change roles at one point, but everyone we were adding was getting the same permissions (i.e. need to edit data in the field), so we took it out for efficiency.

But the "Limited External Collaborator" @torrin47 mentioned is new since we built the tool, and he has mentioned it in email, just haven't had a chance to implement anything.

As for auth groups, it has def been discussed, but no perfect solution since responses can be such a mess of people from all over. In general, if we created a user for a past R9 response, we assigned them to the R9 auth group. Combined w/ #39, think this still makes sense. We'll be able to look back and know why someone was added to a region. And if someone moves regions and/or really should have been in a diff group, someone can always swap them in the Dashboard later, if needed?

**Comment by tbock** [comment]
this is the main issue but the proposed resolution to this ended up in #41 ?

**Comment by tbock** [comment]
data model reworked to force identification of a response or project for which the account is being requested.  These response/projects have a auth group and role assigned to them behind the scenes for gathering by NGST.

Deployed.  Please review @torrin47

**Comment by torrin47** [comment]
Ok, in the admin interface I was able to correctly assign a default role and groups to a project response.  Seemed great. I went and registered a public user with that response, but when I log in as the coordinator, no groups or sponsors are populated, and I can't assign the response or the groups:

![image](https://user-images.githubusercontent.com/2127036/85798937-60d0ab00-b70c-11ea-87cb-f5fe588640c6.png)

**Comment by tbock** [comment]
Yes that's expected b/c you are not assigned to a response.   However, it just dawned on me that your dashboard account will need to see all responses.  I did that for the groups last time so just need to rework that code to do the same with responses.

**Comment by tbock** [comment]
Can we close this?

**Comment by torrin47** [comment]
Yup.
